### PR TITLE
Added initial support for gcc's gomp implementation.

### LINKED ISSUE #1: Seems to be an empty module
Doesn't seem to actually implement the methods described in documentation:

```
% perl -MAlien::OpenMP -E 'say Alien::OpenMP->cflags'
Can't locate object method "cflags" via package "Alien::OpenMP" at -e line 1.
```

### PR DESCRIPTION
**Pull request by oodler577**
cflags   -> -fopenmp
  ldflags  -> -fopenmp

Makefile.PL also uses Devel::CheckLib to verify we're
in a sane gcc environment when ccname == 'gcc'.

### CODE REVIEW

**Comment by mohawk2** [comment]
This is a great start! A few points:
- Devel::CheckLib will need to be a `CONFIGURE_REQUIRES`
- the CI will need to pass, and probably the Ubuntu package that needs adding is `libomp-dev`
- it's not acceptable for the module to install (or indeed succeed in the configure phase) if the environment can't do OpenMP, so the `if gcc` bit (which obviously can be expanded for other possibilities) needs an `else` that does `os_unsupported`
- the `check_lib` bit will need a `function` as previously discussed to ensure actual OpenMP per the standard (that could go in the `else` as a Hail Mary) - this implies starting with a `%check_lib_args` of defaults which get added to in the `if` branches, then `check_lib` gets run once with it (achieving DRY)
- the test shows a code-driven approach, and a more sustainable one is to separate the environment-info-gathering phase, from the info-interpretation phase; with that, a test could run anywhere since it would just have a hardcoded list of infos, and do `is_deeply` on the interpretations

**Comment by oodler577** [comment]
Note on one of your comments above, @mohawk2 -

> the test shows a code-driven approach, and a more sustainable one is to separate the environment-info-gathering phase, from the info-interpretation phase; with that, a test could run anywhere since it would just have a hardcoded list of infos, and do is_deeply on the interpretations

If you are suggesting maintaining a set of profiles, I'd prefer to way until we might actually need it. The current approach simply updates a localized `%Config::Config` by inserting `compiler_name` as desired. I am open to expanding this in either case. The test is meant to be decoupled from the environment. If you can show me how it's not, I'd be happy make sure that it's fully independent as such. Since I do agree with your main point, which I read as, "the tests should run everywhere regardless of the environment."

**Review by mohawk2 — CHANGES_REQUESTED** [review]
I don't think all the points mentioned in my comment on the main PR have been addressed yet? Further thoughts:
- An example to copy for installing Ubuntu packages: https://github.com/PDLPorters/perl5-PGPLOT/blob/master/.github/workflows/ci.yml#L45
- It makes a reviewer's life easier if you avoid unnecessary whitespace diffs

**Comment by oodler577** [comment]
still working down the list:)

**Comment by oodler577** [comment]
@mohawk2 I expect the changes to pass. I believe I have addressed all issues you presented. Please let me know if there is anything else. I'd like to add you under 'AUTHORS', please tell me how you'd like that listed as one final change.

**Comment by oodler577** [comment]
@mohawk2 - no rush, but I'd like to get these changes finalized today. I am eager to begin using this modules for other things. I also have quite a few "Alien" modules I think I might be able to knock out for (HPC related mostly). Anyway, just wanted to finish so I can recover any bandwidth and even utlize this for the talk I am preparing for the end of the month. Thanks for your continued guidence!

File: lib/Alien/OpenMP.pm [hunk 1]
```diff
@@ -15,15 +43,80 @@ Alien::OpenMP - Encapsulate system info for OpenMP
...
+If used for an unsupported compiler, C<ExtUtils::MakeMaker::os_unsupported> is
+invoked, which results an exception propagating from this method being raised
+with the value of C<qq{OS unsupported\n}> (note the new line).
+
+This module assumes that the compiler in question is the same one used to
+build C<perl>. Since the vast majority of C<perl>s are building using C<gcc>,
+initial support is targeting it. However, like C<perl>, many other compilers
+may be used.
+
+Adding support for a new compiler should be straightforward; while C<pull
```
Comment (mohawk2) [inline lib/Alien/OpenMP.pm]: I'd have thought this should be an `I<>` or nothing at all, it's not code? :-)

File: lib/Alien/OpenMP.pm [hunk 2]
```diff
@@ -15,15 +43,80 @@ Alien::OpenMP - Encapsulate system info for OpenMP
...
+
+    gcc -fopenmp ./my-openmp.c -o my-openmp.x
+
+=back
+
+=head2 Note On Compiler Support
+
+If used for an unsupported compiler, C<ExtUtils::MakeMaker::os_unsupported> is
+invoked, which results an exception propagating from this method being raised
+with the value of C<qq{OS unsupported\n}> (note the new line).
```
Comment (mohawk2) [inline lib/Alien/OpenMP.pm:123]: This is not important detail. The user notionally has to read it, or skim it to learn if they can ignore it. Also, you had to write it. What does it add that "won't install if not valid" didn't?

File: t/01-basic.t [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# 'ccname' that gets added to Alien::OpenMP; please
+# don't get fancy and create a loop
+
+# dev note 2: this test is not dependent on the
+# environment, see the use of a local'd version of
+# %Config::Config below for a hint on how to properly
+# add additional tests as compiler support is added to
+# this module.
+
+GCC:
```
Comment (mohawk2) [inline t/01-basic.t:25]: My suggestion was of a separate phases of info-gather and info-interpret, which would then be testable with `is_deeply`. This is still code.

File: xt/pod.t [hunk 4]
```diff
@@ -1,4 +1,6 @@
-use strict; use warnings;
+use strict;
```
Comment (mohawk2) [inline xt/pod.t]: What is the benefit of changing this? A reviewer has to read through these diffs before being able to conclude they are semantically null.

File: Makefile.PL [hunk 5]
```diff
@@ -1,62 +1,99 @@
...
+# check for the library, head, and flags needed to use OpenMP
+if ( q{gcc} eq $Config::Config{ccname} ) {
+    local $/ = undef;          # slurp mode for main() body in __DATA__
+    check_lib_or_exit(
+        lib     => q{gomp},
+        header  => q{omp.h},
+        ccflags => q{-fopenmp},
+        ldflags => q{-fopenmp},
+        function => <DATA>,    # main() body defined in __DATA__
+    );
```
Comment (mohawk2) [inline Makefile.PL]: This is still not a `%check_lib_args` defined above the `if` block, modified within the `if` block, and one `check_lib_or_exit` call in the whole program. What would the code look like if you supported a second environment?

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -46,7 +50,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [macos-latest, windows-latest]
+        os: [windows-latest]
```
Comment (mohawk2) [inline .github/workflows/ci.yml:50]: Is there a way to also support MacOS? See https://stackoverflow.com/questions/43555410/enable-openmp-support-in-clang-in-mac-os-x-sierra-mojave

**Review by mohawk2 — COMMENTED** [review]
> Note on one of your comments above, @mohawk2 -
> > the test shows a code-driven approach, and a more sustainable one is to separate the environment-info-gathering phase, from the info-interpretation phase; with that, a test could run anywhere since it would just have a hardcoded list of infos, and do is_deeply on the interpretations
> 
> If you are suggesting maintaining a set of profiles, I'd prefer to way until we might actually need it. The current approach simply updates a localized `%Config::Config` by inserting `compiler_name` as desired.

I can see it does that. I am advocating you change it.

> I am open to expanding this in either case. The test is meant to be decoupled from the environment. If you can show me how it's not, I'd be happy make sure that it's fully independent as such. Since I do agree with your main point, which I read as, "the tests should run everywhere regardless of the environment."

The situation so far is that indeed is! All we're discussing is how. I have yet to see a situation in computing where code-driven is better than data-driven; nearly all situations I've seen have had data-driven approaches be much easier, and much better.

File: .github/workflows/ci.yml (hunk 6 above)
Comment (oodler577) [reply .github/workflows/ci.yml:50]: yes, will do

File: Makefile.PL (hunk 5 above)
Comment (oodler577) [reply Makefile.PL]: Oh, you're saying to set it up to be easily extended in the future. Will do, now that I understand what you mean :)

**Comment by mohawk2** [comment]
There is still no separate info-gather and info-evaluating phases which would enable the "basic" test to be data-driven. I will merge and release this anyway.

**Pull request merged**